src/datastore/pg-write-store.ts

```diff
@@ -1700,17 +1717,47 @@ export class PgWriteStore extends PgStore {
     return txIds;
   }
 
+  private _debounceMempoolStat: {
+    lastTrigger?: number | null;
+    debounce?: NodeJS.Timeout | null;
+    running: boolean;
+  } = { running: false };
+  /**
+   * Debounce the mempool stat process in case new transactions pour in.
+   */
+  private debounceMempoolStat() {
+    this._debounceMempoolStat.lastTrigger = Date.now();
+    if (this._debounceMempoolStat.running) return;
+    if (this._debounceMempoolStat.debounce != null) {
+      clearTimeout(this._debounceMempoolStat.debounce);
+      this._debounceMempoolStat.debounce = null;
+    }
+    this._debounceMempoolStat.debounce = setTimeout(async () => {
+      this._debounceMempoolStat.running = true;
+      this._debounceMempoolStat.lastTrigger = null;
+      try {
+        const mempoolStats = await this.getMempoolStatsInternal({ sql: this.sql });
+        this.eventEmitter.emit('mempoolStatsUpdate', mempoolStats);
+      } catch (e) {
+        logger.error(e, `failed to reconcile mempool`);
+      } finally {
+        this._debounceMempoolStat.running = false;
+        if (this._debounceMempoolStat.lastTrigger != null) {
+          this.debounceMempoolStat();
+        }
+      }
+    }, MEMPOOL_STATS_DEBOUNCE_INTERVAL);
```

Comment:
Should we be concerned about a possible very long delay in the case of receiving a new mempool tx at least once per MEMPOOL_STATS_DEBOUNCE_INTERVAL for a prolonged period of time? Could be solvable by having another timeout for "max stats wait period" where it runs the stats if too much time has passed?

Reply:
Added the logic.